src/setup-elixir.js

```diff
@@ -73,22 +105,20 @@ function getRunnerOSVersion() {
   return mapToUbuntuVersion[process.env.ImageOS] || 'ubuntu-18.04'
 }
 
-exports.getElixirVersion = getElixirVersion
-
 async function getElixirVersion(spec, otpVersion) {
   const versions = await getElixirVersions()
   const semverRegex = /^v(\d+\.\d+\.\d+(?:-.+)?)/
 
   const semverVersions = Array.from(versions.keys())
-    .filter(str => str.match(semverRegex))
-    .map(str => str.match(semverRegex)[1])
+    .filter((str) => str.match(semverRegex))
+    .map((str) => str.match(semverRegex)[1])
 
   const version = getVersionFromSpec(spec, semverVersions)
   const gitRef = version ? `v${version}` : spec
   const otpMatch = otpVersion.match(/^OTP-([\.\d]+)/)
 
-  if (otpMatch != null && versions.get(gitRef).includes(otpMatch[0])) {
```

Comment:
This one's odd: found it after fixing the tests for local deployment.
The input `otpVersion` has to have `OTP-` otherwise `otpMatch` is null (tests fail).
On the other hand, if that's the case, `otpMatch[0]` will contain the `OTP-` prefix.

**Note**: I can only assume this was dead code.

Reply:
This was added by @ericm a few months ago, it surely must have been added for a reason.  It looks like it's just being defensive and that makes sense to me for the case. 

https://github.com/erlef/setup-elixir/commit/4edd0cd12d3a1a7052d40c1e51d669262e3c7b3b

Reply:
... which is why I find it odd 😄 

`versions.get(gitRef)`, as is, will always fail to include `otpMatch[0]`; with `[1]` though I expect to fix the behaviour as to what we need/want. 

@ericmj?

Reply:
If you are changing it can you add a test for it to verify that it works?

Reply:
Well, the tests did fail because of it (but they weren't running in CI, as explained [here](https://github.com/erlef/setup-elixir/issues/8#issue-831022789): test.yml contains npm test (that's "failing", even though not signalling CI properly)). Which is why it was changed. Want me to create a more low level unit test?

Reply:
It definitely would not hurt @paulo-ferraz-oliveira

Reply:
Sure, but let me rephrase: as-is the code that was changed IS being tested. It wasn't previously, which is what the tests (the fact that `npm test` started being used) actually helped find/fix.

Reply:
I see. If it's covered, then it's covered IMO.

Reply:
I added more tests (with OTP-, without OTP, failing installations, and such) so we get better coverage.
I also added `core.error` and `core.warning` where I found appropriate to ease consumption of the action.